File: clients/metorial-js/packages/sdk/src/sdk.ts
import {
  MetorialFilesEndpoint,
  MetorialInstanceEndpoint,
  MetorialLinksEndpoint,
  MetorialSecretsEndpoint,
  MetorialServerRunErrorsEndpoint,
  MetorialServerRunsEndpoint,
  MetorialServersCapabilitiesEndpoint,
  MetorialServersDeploymentsEndpoint,
  MetorialServersEndpoint,
  MetorialServersImplementationsEndpoint,
  MetorialServersVariantsEndpoint,
  MetorialServersVersionsEndpoint,
  MetorialSessionsEndpoint,
  MetorialSessionsEventsEndpoint,
  MetorialSessionsMessagesEndpoint,
  MetorialSessionsServerSessionsEndpoint
} from '@metorial/generated';
import { MetorialKeyPrefix, sdkBuilder } from './builder';

export let createMetorialSDK = sdkBuilder.build(
  (soft: {
    apiKey: `${MetorialKeyPrefix}${string}` | string;
    apiVersion?: '2025-01-01-pulsar';
    clientSecret?: string;
    asUserId?: string;
    headers?: Record<string, string>;
    apiHost?: string;
  }) => ({
    ...soft,
    apiVersion: '2025-01-01-pulsar'
  })
)(manager => ({
  instance: new MetorialInstanceEndpoint(manager),

  secrets: new MetorialSecretsEndpoint(manager),

  servers: Object.assign(new MetorialServersEndpoint(manager), {
    variants: new MetorialServersVariantsEndpoint(manager),
    versions: new MetorialServersVersionsEndpoint(manager),

    deployments: new MetorialServersDeploymentsEndpoint(manager),
    implementations: new MetorialServersImplementationsEndpoint(manager),

    errors: Object.assign(new MetorialServerRunErrorsEndpoint(manager)),

    runs: new MetorialServerRunsEndpoint(manager),

    capabilities: new MetorialServersCapabilitiesEndpoint(manager)
  }),

  sessions: Object.assign(new MetorialSessionsEndpoint(manager), {
    events: new MetorialSessionsEventsEndpoint(manager),
    messages: new MetorialSessionsMessagesEndpoint(manager),
    serverSessions: new MetorialSessionsServerSessionsEndpoint(manager)
  }),

  files: Object.assign(new MetorialFilesEndpoint(manager), {
    links: new MetorialLinksEndpoint(manager)
  })
}));

export type MetorialSDK = ReturnType<typeof createMetorialSDK>;

File: clients/metorial-py/packages/sdk/sdk.py
from sdk_builder import MetorialSDKBuilder
from metorial_util_endpoint import MetorialEndpointManager
from mt_2025_01_01_pulsar.endpoints.sessions import MetorialSessionsEndpoint

def get_config(soft):
    return {
        **soft,
        "apiVersion": soft.get("apiVersion", "2025-01-01-pulsar"),
    }

def get_headers(config):
    return {"Authorization": f"Bearer {config['apiKey']}"}

def get_api_host(config):
    return config.get("apiHost", "https://api.metorial.com")


def get_endpoints(manager: MetorialEndpointManager):
    return {
        "sessions": MetorialSessionsEndpoint(manager),
        # Add other endpoints here
    }

def create_metorial_sdk():
    builder = MetorialSDKBuilder.create("myapi", "2025-01-01-pulsar") \
        .set_get_api_host(get_api_host) \
        .set_get_headers(get_headers)
    return builder.build(get_config)(get_endpoints)